File: backend/app/schemas/voice.py
"""
app/schemas/voice.py

Response shape for POST /api/v1/voice/transcribe. Owned by: Member 3 —
Application lifecycle & Voice.
"""

from typing import Literal

from pydantic import BaseModel

SUPPORTED_LANGUAGES = ("en", "hi", "mr", "ta", "te", "kn", "ml", "bn", "gu", "pa")
VoiceLanguage = Literal["en", "hi", "mr", "ta", "te", "kn", "ml", "bn", "gu", "pa"]


class TranscribeOut(BaseModel):
    """Response of POST /voice/transcribe. `confidence` is derived from
    faster-whisper's per-segment avg_logprob (converted to a 0-1-ish score),
    not a calibrated probability — good enough to flag "this transcription is
    probably garbage" in the UI, not for anything more precise than that."""

    text: str
    language: str
    duration: float
    confidence: float

File: frontend/lib/voice.ts
/**
 * Client binding for MY (Member 3) POST /voice/transcribe endpoint. See
 * backend/app/api/v1/voice.py and backend/app/schemas/voice.py.
 */

import { apiPostForm } from "@/lib/api-client"

export const SUPPORTED_VOICE_LANGUAGES = ["en", "hi", "mr", "ta", "te", "kn", "ml", "bn", "gu", "pa"] as const
export type VoiceLanguage = (typeof SUPPORTED_VOICE_LANGUAGES)[number]

export interface TranscribeOut {
  text: string
  language: string
  duration: number
  confidence: number
}

export function transcribeAudio(audio: Blob, lang: VoiceLanguage): Promise<TranscribeOut> {
  const formData = new FormData()
  formData.append("file", audio, "recording.webm")
  formData.append("lang", lang)
  return apiPostForm<TranscribeOut>("/voice/transcribe", formData)
}
